frontend/src/components/CalendarView/CalendarView.jsx

```diff
@@ -72,11 +73,19 @@ function CalendarView () {
           <p>Location: {calEvent.location}</p>
         </>
       )
-    })
+    });
+
+    const handleParsedEvents = (parsedEvents) => {
+      parsedEvents.forEach(event => {
+        createCalendarEvent(event);
```

Comment:
JC suggested to move this function into a separate file. When you do this, the ICSUpload component will not need the prop in line 88, you can call it directly in it's own component.